src/dsl.jl

```diff
@@ -110,159 +50,186 @@ end
 end
 
-# Structure containing information about one Reaction. Contain all its substrates and products as well as its rate. Contains a specialized constructor.
-struct ReactionStruct
-    substrates::Vector{ReactantStruct}
-    products::Vector{ReactantStruct}
+# Internal structure containing information about one Reaction. Contain all its substrates and
+# products as well as its rate and potential metadata. Uses a specialized constructor.
+struct DSLReaction
+    substrates::Vector{DSLReactant}
+    products::Vector{DSLReactant}
     rate::ExprValues
     metadata::Expr
     rxexpr::Expr
 
-    function ReactionStruct(sub_line::ExprValues, prod_line::ExprValues, rate::ExprValues,
-            metadata_line::ExprValues, rx_line::Expr)
-        sub = recursive_find_reactants!(sub_line, 1, Vector{ReactantStruct}(undef, 0))
-        prod = recursive_find_reactants!(prod_line, 1, Vector{ReactantStruct}(undef, 0))
+    function DSLReaction(sub_line::ExprValues, prod_line::ExprValues,
+            rate::ExprValues, metadata_line::ExprValues, rx_line::Expr)
+        subs = recursive_find_reactants!(sub_line, 1, Vector{DSLReactant}(undef, 0))
+        prods = recursive_find_reactants!(prod_line, 1, Vector{DSLReactant}(undef, 0))
         metadata = extract_metadata(metadata_line)
-        new(sub, prod, rate, metadata, rx_line)
+        new(subs, prods, rate, metadata, rx_line)
     end
 end
 
 # Recursive function that loops through the reaction line and finds the reactants and their
-# stoichiometry. Recursion makes it able to handle weird cases like 2(X+Y+3(Z+XY)).
+# stoichiometry. Recursion makes it able to handle weird cases like 2(X + Y + 3(Z + XY)). The
+# reactants are stored in the `reactants` vector. As the expression tree is parsed, the
+# stoichiometry is updated and new reactants added.
 function recursive_find_reactants!(ex::ExprValues, mult::ExprValues,
-        reactants::Vector{ReactantStruct})
-    if typeof(ex) != Expr || (ex.head == :escape) || (ex.head == :ref)
+        reactants::Vector{DSLReactant})
+    # We have reached the end of the expression tree and can finalise and return the reactants.
+    if (typeof(ex) != Expr) || (ex.head == :escape) || (ex.head == :ref)
+        # The final bit of the expression is not a relevant reactant, no additions are required.
         (ex == 0 || in(ex, empty_set)) && (return reactants)
-        if any(ex == reactant.reactant for reactant in reactants)
-            idx = findall(x -> x == ex, getfield.(reactants, :reactant))[1]
-            reactants[idx] = ReactantStruct(ex,
-                processmult(+, mult,
-                    reactants[idx].stoichiometry))
+
+        # If the expression corresponds to a reactant on our list, increase its multiplicity.
+        idx = findfirst(r.reactant == ex for r in reactants)
+        if !isnothing(idx)
+            new_mult = processmult(+, mult, reactants[idx].stoichiometry)
+            reactants[idx] = DSLReactant(ex, new_mult)
```

Comment:
Instead of one long line that get split into two lines by the line break I just wrote it out into two lines
```julia
new_mult = processmult(+, mult, reactants[idx].stoichiometry)
reactants[idx] = DSLReactant(ex, new_mult)
```